shashi/feb6.js

```diff
@@ -0,0 +1,37 @@
+// Swapping Two by Two
+// Write a function that swaps the first pair (1st and 2nd characters) with the second pair (3rd and 4th characters) for every quadruplet substring.
+// Examples
+// swapTwo("ABCDEFGH") ➞ "CDABGHEF"
+// swapTwo("AABBCCDDEEFF") ➞ "BBAADDCCFFEE"
+// swapTwo("munchkins") ➞ "ncmuinhks"
+// swapTwo("FFGGHHI") ➞ "GGFFHHI"
+// Notes
+// Keep leftover strings in the same order.
+
+const swapTwo = (string) => {
+  var newStr = "";
+  var remChars = string.length % 4;
+  for (let i = 0; i < (string.length - remChars); i += 4) {
+    newStr += string[i + 2] + string[i + 3] + string[i] + string[i + 1];
+  }
+  if (remChars !== 0) {
+    for (let i = string.length - remChars; i < string.length; i++) {
+      newStr += string[i];
+    }
+  }
+  return newStr;
+}
```

Comment:
LGTM - 

suggest to check string.length % 4 after iteration.

since you're iterating through 4 your loop will stop if the last few characters are not divisible by 4.

[0,1,2,3,4,5,6]  -  i < arr.length; i +=4
first iteration - hits 3. 
next iteration - i = 8 is greater than arr.length. The loop exits.
check %4 here.
if left over iterate through the rest.